### PR TITLE
[GPU][DG2] Fix output scale post-op condition

### PR DESCRIPTION
**Pull request by tuxedcat**
### Related Test Cases:
 - Error Message: could not create a primitive descriptor iterator
   - conv_fp32_scale_quantize_i8.basic/2
   - conv_fp32_scale_activation_quantize_i8_eltwise_fp32_quantize_i8.basic/0
   - deconv_scale_actv_quant_i8.basic/9

### Details:
 - Fix output scale post-op condition to fit onednn

### Tickets:
 - 67491

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/program_node.cpp [hunk 1]
```diff
@@ -989,9 +989,16 @@ void program_node::init_onednn_primitive_attributes() {
...
-                    !(is_type<convolution>() && data_type_traits::is_floating_point(output_layout.data_type))) {
+                auto input_datatype = get_dependency(0).get_output_layout().data_type;
+                // convolution using post-op output scales can only be used when i8/u8 input (which use integer accumulator)
+                bool can_use_output_scales =
+                    idx != 0 ||
+                    has_out_scales(attrs) ||
+                    is_type<pooling>() ||
+                    is_type<reduce>() ||
+                    //deconv is work around. TODO: update desc.type in pre_replace_deconv pass
+                    (is_type<convolution>() || is_type<deconvolution>) && data_type_traits::is_floating_point(input_datatype);
```
[the remark this answers is not in the document]
Comment (tuxedcat) [reply src/plugins/intel_gpu/src/graph/program_node.cpp]: NVM. I thought deconv is optimized but not optimized actually.

File: src/plugins/intel_gpu/src/graph/program_node.cpp [hunk 2]
```diff
@@ -989,16 +989,23 @@ void program_node::init_onednn_primitive_attributes() {
...
+                auto input_datatype = get_dependency(0).get_output_layout().data_type;
+                // convolution using post-op output scales can only be used when i8/u8 input (which use integer accumulator)
+                bool cant_use_output_scales =
+                    idx != 0 ||
+                    has_out_scales(attrs) ||
+                    is_type<pooling>() ||
+                    is_type<reduce>() ||
+                    (is_type<convolution>() && data_type_traits::is_floating_point(input_datatype)) ||
+                    (is_type<deconvolution>() && data_type_traits::is_floating_point(input_datatype));
+                if (cant_use_output_scales) {
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/program_node.cpp:1001]: I think not(!) is required here?

**Review by isanghao — APPROVED** [review]
Looks good to me. Let's check the performance.

File: src/plugins/intel_gpu/src/graph/program_node.cpp (hunk 2 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/src/graph/program_node.cpp:1001]: I've changed if-else code block instead

**Review by isanghao — APPROVED** [review]

**Pull request merged**